src/tree_model/leaf.py

```diff
@@ -1,5 +1,6 @@
+
 class Leaf:
-    def __init__(self, label_value, depth: int):
+    def __init__(self, label_value, depth: int, parent_node: "Node"):
```

Comment:
Don't need "" around Node
```suggestion
    def __init__(self, label_value, depth: int, parent_node: Node):
```